backend/controllers/prisonerProfile/prisonerCellHistory.js

```diff
@@ -0,0 +1,87 @@
+const { serviceUnavailableMessage } = require('../../common-messages')
+const {
+  formatName,
+  stripAgencyPrefix,
+  formatTimestampToDateTime,
+  possessive,
+  sortByDateTime,
+  putLastNameFirst,
+} = require('../../utils')
+const {
+  app: { notmEndpointUrl: dpsUrl },
+} = require('../../config')
+
+module.exports = ({ elite2Api, logError, page = 0 }) => async (req, res) => {
+  const { offenderNo } = req.params
+
+  const getAgencyDetails = async cells => {
+    return Promise.all(
+      cells
+        .map(cell => cell.agencyId)
+        .filter((v, i, a) => a.indexOf(v) === i)
```

Comment:
Make the list unique on agency id. Prevents making multiple calls to the API for the same agency detail

Reply:
Not sure if it's better but the way we've made list unique is by doing: `Array.from(new Set(data.map(event => event.bookingId)))`